sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/SymbolicExecution/RestrictDeserializedTypes.cs

```diff
@@ -322,5 +371,43 @@ private bool HasSafeBinderInitialization(InitializerExpressionSyntax initializer
             private static bool IsLosFormatter(ITypeSymbol typeSymbol) =>
                 typeSymbol.Is(KnownType.System_Web_UI_LosFormatter);
         }
+
+        private sealed class LocationInfo
+        {
+            internal Location Primary { get; }
+
+            internal string Message { get; }
+
+            internal IEnumerable<Location> SecondaryLocations { get; }
+
+            public LocationInfo(Location primary, string message, Location secondary = null)
+            {
+                Primary = primary;
+                Message = message;
+                SecondaryLocations = secondary == null
+                    ? Enumerable.Empty<Location>()
+                    : new[] {secondary};
+            }
+        }
+
+        private sealed class TypeDeclarationInfo
+        {
+            internal Location Location { get; }
+
+            internal bool IsSafe { get; }
+
+            public TypeDeclarationInfo(MethodDeclarationSyntax declaration)
+            {
```

Comment:
if think you can refactor this to
```
if (declaration == null)
{
  Location = null;
  IsSafe = true;
}
else
{
  Location = ..;
  IsSafe = declaration...;
}
```

otherwise, there are two null-checks which makes it harder to read

Reply:
But this would mean to replace 2 lines with 10. In my opinion the code is more readable in the current form.